boards/PDM/Inc/Io/Io_Efuse.h

```diff
@@ -0,0 +1,145 @@
+/**
+ * @brief Driver for interfacing with the 22XS4200 E-Fuse via SPI
+ * @Settings Efuse max SPI transfer rate: 8MHz
+ *           SPI Clock Polarity: 0 (SCK low-level idle state)
+ *           SPI Clock Phase: 2nd edge (slave samples MOSI on SCK falling edge)
+ *           Slave Select: Active Low (Must be toggled between SPI messages)
+ */
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "main.h"
+#include "App_SharedExitCode.h"
+
+struct Efuse_Context;
+
+/**
+ * Allocate and initialize an Efuse
+ * @param hspi Handle to the SPI peripheral used for the Efuse
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_port Handle to Efuse's chip-select GPIO pin
+ * @param fsob_port Handle to Efuse's fail-safe output GPIO port
+ * @param fsob_pin Handle to Efuse's fail-safe output GPIO pin
+ * @param fsb_port Handle to Efuse's fault status GPIO port
+ * @param fsb_pin Handle to Efuse's fault status GPIO pin
+ * @param channel0_port Handle to Efuse's channel 0 GPIO port
+ * @param channel0_pin Handle to Efuse's channel 0 GPIO pin
+ * @param channel1_port Handle to Efuse's channel 1 GPIO port
+ * @param channel1_pin Handle to Efuse's channel 1 GPIO pin
+ * @return The created Efuse, whose ownership is given to the caller
+ */
+struct Efuse_Context *Io_Efuse_Create(
+    SPI_HandleTypeDef *spi_handle,
+    GPIO_TypeDef *     nss_port,
+    uint16_t           nss_pin,
+    GPIO_TypeDef *     fsob_port,
+    uint16_t           fsob_pin,
+    GPIO_TypeDef *     fsb_port,
+    uint16_t           fsb_pin,
+    GPIO_TypeDef *     channel0_port,
+    uint16_t           channel0_pin,
+    GPIO_TypeDef *     channel1_port,
+    uint16_t           channel1_pin);
+
+/**
+ * Deallocate the memory used by the given Efuse
+ * @param e_fuse The Efuse to deallocate
+ */
+void Io_Efuse_Destroy(struct Efuse_Context *efuse);
+
+/**
+ * Enable the given efuse's channel 0 output.
+ */
+void Io_Efuse_EnableChannel0(const struct Efuse_Context *const efuse);
+
+/**
+ * Disable the given efuse's channel 0 output.
+ */
+void Io_Efuse_DisableChannel0(const struct Efuse_Context *const efuse);
+
+/**
+ * Enable the given efuse's channel 1 output.
+ */
+void Io_Efuse_EnableChannel1(const struct Efuse_Context *const efuse);
+
+/**
+ * Disable the given efuse's channel 1 output.
+ */
+void Io_Efuse_DisableChannel1(const struct Efuse_Context *const efuse);
+
+/**
+ * Set the current or temperature monitoring option of the CSNS pin for the
+ * given efuse.
+ * @param monitoring_function The desired monitoring function which can be
+ * selected by specifying the following values: CSNS_FUNCTION_DISABLED - Current
+ * sensing disabled CSNS_FUNCTION_CURRENT_CH0 - Current sensing for Channel 0
```

Comment:
Ah okay i see what you mean about this now. I was misled since AUX1/2_CURRENT_SENSE_CHANNEL were used as inputs to this function. I may have asked you this already but will we use temperature and current sum? Its bin a while

Reply:
I don't think its a bad idea to get rid of the AUX1/2_CURRENT_SENSE_CHANNEL constants and just pass CSNS_FUNCTION_CURRENTs in the Aux1Aux2 file.

Reply:
temperature most likely never, sum could be used in the future but i'm not sure tbh. Got rid of the constants like in your other comment, function is moved into Io_efuse file now